pettingzoo/mpe/_mpe_utils/simple_env.py

```diff
@@ -231,75 +246,80 @@ def step(self, action):
         self._cumulative_rewards[cur_agent] = 0
         self._accumulate_rewards()
 
-    def render(self, mode="human"):
-        from . import rendering
-
-        if self.viewer is None:
-            self.viewer = rendering.Viewer(700, 700)
-
-        # create rendering geometry
-        if self.render_geoms is None:
-            # import rendering only if we need it (and don't import for headless machines)
-            # from gym.envs.classic_control import rendering
-            # from multiagent._mpe_utils import rendering
-            self.render_geoms = []
-            self.render_geoms_xform = []
-            for entity in self.world.entities:
-                geom = rendering.make_circle(entity.size)
-                xform = rendering.Transform()
-                if "agent" in entity.name:
-                    geom.set_color(*entity.color[:3], alpha=0.5)
-                else:
-                    geom.set_color(*entity.color[:3])
-                geom.add_attr(xform)
-                self.render_geoms.append(geom)
-                self.render_geoms_xform.append(xform)
-
-            # add geoms to viewer
-            self.viewer.geoms = []
-            for geom in self.render_geoms:
-                self.viewer.add_geom(geom)
-
-            self.viewer.text_lines = []
-            idx = 0
-            for agent in self.world.agents:
-                if not agent.silent:
-                    tline = rendering.TextLine(self.viewer.window, idx)
-                    self.viewer.text_lines.append(tline)
-                    idx += 1
-
-        alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
-        for idx, other in enumerate(self.world.agents):
-            if other.silent:
-                continue
-            if np.all(other.state.c == 0):
-                word = "_"
-            elif self.continuous_actions:
-                word = "[" + ",".join([f"{comm:.2f}" for comm in other.state.c]) + "]"
-            else:
-                word = alphabet[np.argmax(other.state.c)]
+    def enable_render(self):
+        self.screen = pygame.display.set_mode(self.screen.get_size())
+        self.renderOn = True
+        self.draw()
```

Comment:
The self.draw could be removed from `enable_render` function. That way `enable_render` would have one specific responsibility. But I would include the flag to render or no depends if the mode is human.

```suggestion
    def enable_render(self, mode="human"):
        if not self.renderOn and mode == "human":
            self.screen = pygame.display.set_mode(self.screen.get_size())
            self.renderOn = True
```
If in the future you need to enable render for multiple modes you only changes in one place.

What do you think?

Reply:
Thanks for the compliment! I normally just write code for my eyes only so it means a lot :)

The style of this code was inspired by cooperative_pong, which admittedly I haven't had time to fully understand. I'll quote the code from cooperative_pong below:

```
    def enable_render(self):
        self.screen = pygame.display.set_mode(self.screen.get_size())
        self.renderOn = True
        self.draw()

    def render(self, mode="human"):
        if not self.renderOn and mode == "human":
            # sets self.renderOn to true and initializes display
            self.enable_render()

        observation = np.array(pygame.surfarray.pixels3d(self.screen))
        if mode == "human":
            pygame.display.flip()
        return (
            np.transpose(observation, axes=(1, 0, 2)) if mode == "rgb_array" else None
        )
```

The addition of lines 258-259 was to compensate for the lack of a `self.draw()` call in the `step()` method.

I agree that your refactoring is more elegant, and I would be in favour of these changes. I just thought I'd note that it is a minor divergence from the style of the Butterfly environments, it seems.

Reply:
Having the shorter version of the code is probably ideal, cooperative pong can be changed to reflect this more elegant way of doing things down the line.

Reply:
Nice! Always we can improve to make everything better. In the case of cooperative_pong, the code is quite similar, we can abstract the code and reuse it, but this should be well designed and planned. 
Thanks guys for the discussion.